File: src/models/prompt.ts
/** Types cho thư viện Prompt — khớp với DTO ở `src-tauri/src/models/prompt.rs`. */

export interface Prompt {
  id: number;
  title: string;
  body: string;
  tags: string[];
  category: string;
  usage_count: number;
  created_at: string;
  updated_at: string;
}

export interface PromptRequest {
  title: string;
  body: string;
  tags: string[];
  category: string;
}

/** Tên biến `{{var}}` xuất hiện trong `body` — parse ở frontend, không lưu backend. */
export function extractPromptVariables(body: string): string[] {
  const matches = body.matchAll(/\{\{(\w+)\}\}/g);
  const names = new Set<string>();
  for (const m of matches) names.add(m[1]);
  return [...names];
}

File: src-tauri/src/models/prompt.rs
//! Model cho thư viện Prompt — snippet tái sử dụng, hỗ trợ placeholder dạng
//! `{{variable}}` (parse ở frontend, backend chỉ lưu chuỗi thô).

use serde::{Deserialize, Serialize};

/// Một prompt — thư viện global, tái sử dụng cho mọi workspace/workflow.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct Prompt {
    pub id: i64,
    pub title: String,
    /// Nội dung prompt, có thể chứa placeholder `{{var}}`.
    #[serde(default)]
    pub body: String,
    #[serde(default)]
    pub tags: Vec<String>,
    /// Free-text, không cần enum cứng.
    #[serde(default)]
    pub category: String,
    #[serde(default)]
    pub usage_count: i32,
    pub created_at: String,
    pub updated_at: String,
}

/// Request tạo/cập nhật prompt — dùng chung field cho cả 2.
#[derive(Debug, Deserialize)]
pub struct PromptRequest {
    pub title: String,
    #[serde(default)]
    pub body: String,
    #[serde(default)]
    pub tags: Vec<String>,
    #[serde(default)]
    pub category: String,
}
